vsintegration/src/FSharp.Editor/CodeFixes/FSharpCodeFixContext.fs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+
+open CancellableTasks
+
+type FSharpCodeFixContext(document: Document, span: TextSpan) =
+
+    let mutable _sourceText = None
```

Comment:
@0101 soooo my idea is to cache sourceText here if it was already accessed - is this the right way to go? Yesterday you mentioned you're exercising a lot with caching now :D

Basically, this is to avoid double-accessing source text in an async manner when having code like 
```
    let! sourceText = context.GetSourceTextAsync()
    let! errorRange = context.GetErrorRangeAsync()
```

Reply:
It might be enough to just rely on `Document.GetTextAync` to do this for you. It shouldn't do much extra work if text has already been loaded.

https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/TextDocumentState.cs,e0d44734c1f9b789

Reply:
Yeah I guess I am just reinventing lazy here. Lazy should be fine I think.

Reply:
But even using `lazy` could be reinventing what the `Document` does already. And adding extra allocation and complicating our code for potentially no benefit. Would be nice if we could measure it somehow.

Reply:
Document just puts it into a ValueTask (which is a struct) in case it was already loaded - that should be already friendly enough for CancellableTask without needing any additional wrapper(s).